python/ray/data/_internal/block_batching/block_batching.py

```diff
@@ -98,15 +101,17 @@ def batch_block_refs(
     else:
         prefetcher = WaitBlockPrefetcher()
 
-    block_iter = _resolve_blocks(
+    eager_free = clear_block_after_read and DatasetContext.get_current().eager_free
+
+    block_iter = resolve_block_refs(
         _prefetch_blocks(
```

Comment:
I'm assuming that this is going to be superseded by the new `prefetch_batches_locally` in the third PR?

Reply:
No, the legacy codepath will still use the old `_prefetch_blocks`